Build/Makefile.soc.gd32vf103

```diff
@@ -9,7 +9,16 @@ GD32VF103_SDK_SOC_BOARD=$(NUCLEI_SDK_SOC)/Board/$(BOARD)
 GD32VF103_SDK_SOC_COMMON=$(NUCLEI_SDK_SOC)/Common
 
```

Comment:
Add `VARIANT ?= ` here to let user know there is a new variable

Reply:
My proposal changes:

~~~makefile
BOARD ?= gd32vf103v_rvstar
# Variant for Board or SoC
VARIANT ?=

# override DOWNLOAD and CORE variable for GD32VF103 SoC
# even though it was set with a command argument
override CORE := n205
override DOWNLOAD := flashxip

GD32VF103_SDK_SOC_BOARD=$(NUCLEI_SDK_SOC)/Board/$(BOARD)
GD32VF103_SDK_SOC_COMMON=$(NUCLEI_SDK_SOC)/Common

# no ilm on gd32vf103 SoC
ifeq ($(BOARD), gd32vf103c_longan_nano)
ifeq ($(VARIANT), lite)
LINKER_SCRIPT ?= $(GD32VF103_SDK_SOC_BOARD)/Source/GCC/gcc_gd32vf103x8_flashxip.ld
else
LINKER_SCRIPT ?= $(GD32VF103_SDK_SOC_BOARD)/Source/GCC/gcc_gd32vf103xb_flashxip.ld
endif
else
LINKER_SCRIPT ?= $(GD32VF103_SDK_SOC_BOARD)/Source/GCC/gcc_gd32vf103_flashxip.ld
endif

# Default openocd configuration file used by this SoC
OPENOCD_CFG ?= $(GD32VF103_SDK_SOC_BOARD)/openocd_gd32vf103.cfg

CORE_UPPER = $(call uc, $(CORE))

RISCV_ARCH ?= rv32imac
RISCV_ABI ?= ilp32

# Extra openocd and gdb commands
GDB_UPLOAD_CMDS += -ex "monitor halt"
~~~